Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/UserMetadataServiceImpl.java

```diff
@@ -141,6 +142,42 @@ public void saveUserMetadata(String userMetadataContent, UserMetadataTypeEnum us
 
     }
 
+    /**
+     *
+     * Find UserMetadata that linked to specified Log and Group
+     * @param logId logId
+     * @param groupRowGuid GroupId
+     * @return Set of GroupUsermetadata
+     */
+    private  Set<GroupUsermetadata> findByLogAndGroup(Integer logId, String groupRowGuid) {
+
+        Group group = groupRepo.findByRowGuid(groupRowGuid);
+        Set<GroupUsermetadata> result = new HashSet<>();
+
+        // All the user metadata that linked to this log
+        Set<UsermetadataLog> usermetadataLogSet =
+                new HashSet<>(usermetadataLogRepo.findByLog(logRepo.findUniqueByID(logId)));
+
+        if (usermetadataLogSet.size() != 0) {
+
+            for (UsermetadataLog usermetadataLog : usermetadataLogSet) {
+                Usermetadata u = usermetadataLog.getUsermetadata();
```

Comment:
Not important now, but a quick way to do this is get all the Ids in 1 db quiery, and then use in query to get all list of groupmeta data , and then so the needful.

Reply:
What happens if an arbitrary groupRowGuid value is passed to the findByLogAndGroup() method and a null is returned at line 154?